pkg/supervisor/supervisor_test.go

```diff
@@ -94,16 +94,14 @@ func TestSupervisorStart(t *testing.T) {
 }
 
 func TestGetEnv(t *testing.T) {
-	// backup environment vars, and restore them when test finishes
+	// Cleanup the environment variables before the test
 	oldEnv := os.Environ()
-	t.Cleanup(func() {
-		for _, e := range oldEnv {
-			key, val, _ := strings.Cut(e, "=")
-			assert.NoError(t, os.Setenv(key, val))
-		}
-	})
+	for _, e := range oldEnv {
+		key, _, _ := strings.Cut(e, "=")
+		t.Setenv(key, "")
+		os.Unsetenv(key)
```

Comment:
I see what you did there, but I think it's a bit hard to grasp. Would you mind to add a comment as to why the t.Setenv call is there?

OTOH, I have seen this test pop up in linter runs more than once. It might be worthwhile to nuke the dependency on `os.Environ` in the function being tested and pass the environment as a string slice argument instead. This would allow the above code to be removed.

Reply:
I added a comment, I'm not opposed to the idea of passing a string slice with the environ though.